src/webgpu/api/validation/state/device_lost/destroy.spec.ts

```diff
@@ -454,6 +454,156 @@ Tests creating render pipeline on destroyed device.
     }, awaitLost);
   });
 
+g.test('createComputePipelineAsync')
+  .desc(
+    `
+Tests creating compute pipeline asynchronously on destroyed device.
+  - Tests with a valid no-op compute shader
+  `
+  )
+  .params(u => u.beginSubcases().combine('awaitLost', [true, false]))
+  .fn(async t => {
+    const { awaitLost } = t.params;
+    const cShader = t.device.createShaderModule({ code: t.getNoOpShaderCode('COMPUTE') });
+    const fn = () =>
+      t.device.createComputePipelineAsync({
+        layout: 'auto',
+        compute: { module: cShader, entryPoint: 'main' },
+      });
+
+    // Normal creation succeeds.
+    await fn();
+
+    // Destroy the device, and expect it to be lost.
+    t.expectDeviceLost('destroyed');
+    t.device.destroy();
+    if (awaitLost) {
+      const lostInfo = await t.device.lost;
+      t.expect(lostInfo.reason === 'destroyed');
+    }
+
+    // After device destroy, expect a GPUPipelineError with "validation" reason.
```

Comment:
@kainino0x Want to confirm with you whether this is the intended behavior? W.r.t the other APIs, I think that this should actually just return a pipeline? If so, maybe the `during` variants can be combined.

Reply:
The intent is that it returns a pipeline, not rejects. Although reading the spec closely I think the spec is actually wrong and says it'll reject. Reopened https://github.com/gpuweb/gpuweb/issues/1629#issuecomment-1599988786

Reply:
I've updated these tests to now expect success for all the destroyed cases